core/src/main/java/hudson/model/ManageJenkinsAction.java

```diff
@@ -35,7 +35,7 @@
 @Extension(ordinal=100) @Symbol("manageJenkins")
 public class ManageJenkinsAction implements RootAction {
     public String getIconFileName() {
-        if (Jenkins.get().hasPermission(Jenkins.MANAGE))
+        if (Jenkins.get().hasPermission(Jenkins.MANAGE) || Jenkins.get().hasPermission(Jenkins.SYSTEM_READ))
```

Comment:
could we instead add 2 new methods to `AccessControlled`  `hasAnyPermission(Permission... permission)` / `checkAnyPermission(Permission ... permission)`?  

that would also make https://github.com/jenkinsci/jenkins/pull/4506/files#diff-841df66889e3c3214c68de134dfe05beR1114 more elegant

Reply:
> that would also make https://github.com/jenkinsci/jenkins/pull/4506/files#diff-841df66889e3c3214c68de134dfe05beR1114 more elegant

It wouldn't because Jelly doesn't like varargs. That was my first try.

Reply:
I also gave varargs a go and wasn’t able to make it work

Reply:
improving this anyway, there's a special method in functions for jelly, the rest of the code can be nice

Reply:
s/vararg/array/ ?  or can you not do arrays in jelly either?

Reply:
jelly can do an array (which is what's currently being done), but not varargs

Reply:
so can we have 2 new methods to `AccessControlled` `hasAnyPermission(Permission[] permissions) / `checkAnyPermission(Permission[] permissions)`?

Strange jelly can not do varargs - you can call a vararg method with an array (in java) if that enables a single varags method rather than an array based one.

Reply:
https://github.com/jenkinsci/jenkins/pull/4506/files#diff-416282a0bfa144ea94854dff20c12b67R82-R134

those methods are there,

Yes I know it's weird, DB tried it first, and then I also tried =/

Reply:
> Strange jelly can not do varargs

IIRC it was even stranger: It worked (i.e. correct result wrt permissions) but spewed exceptions.